frontend/src/components/TeledeclarationPreview.vue

```diff
@@ -84,7 +84,32 @@ export default {
         this.$emit("input", newValue)
       },
     },
+    isCentralCanteen() {
+      return this.canteen.productionType === "central_serving" || this.canteen.productionType === "central"
+    },
+    centralKitchenDiagostic() {
+      if (this.diagnostic.year && this.canteen?.centralKitchenDiagnostics)
+        return this.canteen.centralKitchenDiagnostics.find((x) => x.year === this.diagnostic.year)
+      return null
+    },
+    showApproKeys() {
+      if (this.canteen?.productionType === "site_cooked_elsewhere" && this.centralKitchenDiagostic) {
+        return (
+          this.centralKitchenDiagostic.centralKitchenDiagnosticMode !== "APPRO" &&
+          this.centralKitchenDiagostic.centralKitchenDiagnosticMode !== "ALL"
+        )
+      }
+      return true
+    },
+    showAdditionalKeys() {
+      if (this.canteen?.productionType === "site_cooked_elsewhere" && this.centralKitchenDiagostic)
+        return this.centralKitchenDiagostic.centralKitchenDiagnosticMode !== "ALL"
+      if (this.isCentralCanteen)
+        return this.diagnostic.centralKitchenDiagnosticMode && this.diagnostic.centralKitchenDiagnosticMode !== "APPRO"
+      return true
```

Comment:
I find this logic hard to read. Maybe because of the lack of if/else, the default return true, checks that things exist and inverse checks on the values... What are the natural language rules for showing the additional keys ?